client/src/leap/soledad/client/interfaces.py

```diff
@@ -19,6 +19,37 @@
 """
 from zope.interface import Interface, Attribute
 
+#
+# Plugins
+#
+
+
+class ISoledadPostSyncPlugin(Interface):
+    """
+    I implement the minimal methods and attributes for a plugin that can be
+    called after a soledad synchronization has ended.
+    """
+
+    def process_received_docs(self, doc_id_list):
+        """
+        Do something with the passed list of doc_ids received after the last
+        sync.
+
+        :param doc_id_list: a list of strings for the received doc_ids
+        """
+
+    watched_doc_types = Attribute("""
+        a tuple of the watched doc types for this plugin. So far, the
+        `doc-types` convention is just the preffix of the doc_id, which is
+        basically its first character, followed by a dash. So, for instance,
+        `M-` is used for meta-docs in mail, and `F-` is used for flag-docs in
+        mail. For now there's no central register of all the doc-types
+        used.""")
```

Comment:
(reopening the comment thread as the previous got folded by the changes)

I thought this https://github.com/kalikaneko/leap_mail/commit/ee4bb4117c202770df7615f2623032b2021b2a65#diff-0ddf4e813820f3c8cb170b906d84a1baR49 will live in soledad not in each plugin. If soledad don't do anything with `watched_doc_types` I don't see the point of having it in the interface, each plugin could implement it as it wishes.